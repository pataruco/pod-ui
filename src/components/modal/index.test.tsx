import { shallow } from 'enzyme';
import React from 'react';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import Modal, { ModalComponent } from './index';

describe('ModalComponent', () => {
  it('renders', () => {
    const defaultProps = {
      isModalOpen: true,
      date: 'somedate',
      files: [],
    };
    expect(shallow(<ModalComponent {...defaultProps} />)).toMatchSnapshot();
  });
});

describe('Modal', () => {
  const store = configureStore([])({});

  it('renders', () => {
    const defaultProps = {
      isModalOpen: true,
      date: 'somedate',
      files: [],
    };
    expect(
      shallow(
        <Provider store={store}>
          <Modal {...defaultProps} />
        </Provider>,
      ),
    ).toMatchSnapshot();
  });
});
