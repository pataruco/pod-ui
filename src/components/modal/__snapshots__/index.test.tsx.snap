// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal renders 1`] = `
<Connect(ModalComponent)
  date="somedate"
  files={Array []}
  isModalOpen={true}
/>
`;

exports[`ModalComponent renders 1`] = `
<Modal
  ariaHideApp={true}
  bodyOpenClassName="ReactModal__Body--open"
  closeTimeoutMS={0}
  contentLabel="Example Modal"
  isOpen={true}
  onRequestClose={[Function]}
  parentSelector={[Function]}
  portalClassName="ReactModalPortal"
  shouldCloseOnEsc={true}
  shouldCloseOnOverlayClick={true}
  shouldFocusAfterRender={true}
  shouldReturnFocusAfterClose={true}
>
  <button
    onClick={[Function]}
  >
    close
  </button>
  <Carrousel
    date="somedate"
    files={Array []}
  />
</Modal>
`;
